packages/block-library/src/navigation/edit/index.js

```diff
@@ -303,6 +337,53 @@ function Navigation( {
 		// with the snapshot from the time when ref became undefined.
 	}, [ clientId, ref, innerBlocks ] );
 
+	useEffect( () => {
+		const setPermissionsNotice = () => {
+			if ( noticeRef.current ) {
+				return;
+			}
+
+			noticeRef.current = uniqueId( 'navBlockNoEditPermissions' );
+
+			createWarningNotice(
+				__(
+					'You do not have permission to edit this Navigation. Any edits made will not be saved.'
+				),
+				{
+					id: noticeRef.current,
```

Comment:
I think a unique id is overkill for this. It just has to be a string that isn't used elsewhere to avoid multiple notifications with the same message. I think you can make it a namespaced string like 'block-library/core/navigation/permissions'.

Reply:
OK fixed.